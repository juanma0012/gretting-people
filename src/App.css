.App {
  text-align: center;
}
.App > h1 {
  color: #535353;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
