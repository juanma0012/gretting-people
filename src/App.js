import React, { Component } from 'react';
import { PeopleForm, GreetingList, LanguageDropdown } from './components';
import './app.css';

class App extends Component {
    title = 'Greeting people';
    render() {
        return (
            <div className="app">
                <h1>{this.title}</h1>
                <br></br>
                <PeopleForm />
                <br></br>
                <LanguageDropdown />
                <br></br>
                <GreetingList />
            </div>
        );
    }
}

export default App;
