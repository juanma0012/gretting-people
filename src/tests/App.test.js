import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('Component App ', () => {
    describe('WHEN is initialized ', () => {
        const component = shallow(<App />);
        it('SHOULD render correctly ', () => {
            expect(component).toMatchSnapshot();
        });
        it('SHOULD contains a title ', () => {
            expect(component.find('h1').text()).toEqual('Greeting people');
        });
        it('SHOULD contains the PeopleForm component', () => {
            expect(component.find('Connect(PeopleForm)')).toHaveLength(1);
        });
        it('SHOULD contains the LanguageDropdown component', () => {
            expect(component.find('Connect(LanguageDropdown)')).toHaveLength(1);
        });
        it('SHOULD contains the GreetingList component', () => {
            expect(component.find('Connect(GreetingList)')).toHaveLength(1);
        });
    });
});
