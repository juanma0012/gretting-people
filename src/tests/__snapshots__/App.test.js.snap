// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component App  WHEN is initialized  SHOULD render correctly  1`] = `
<div
  className="app"
>
  <h1>
    Greeting people
  </h1>
  <br />
  <Connect(PeopleForm) />
  <br />
  <Connect(LanguageDropdown) />
  <br />
  <Connect(GreetingList) />
</div>
`;
